front/pages/w/[wId]/members/index.tsx

```diff
@@ -56,66 +70,315 @@ export default function WorkspaceAdmin({
+                )}
 
-  const handleUpdateWorkspace = async () => {
-    setUpdating(true);
-    const res = await fetch(`/api/w/${owner.sId}`, {
-      method: "POST",
-      headers: {
-        "Content-Type": "application/json",
-      },
-      body: JSON.stringify({
-        allowedDomain: allowedDomain,
-      }),
-    });
-    if (!res.ok) {
-      window.alert("Failed to update workspace.");
-      setUpdating(false);
-    } else {
-      // We perform a full refresh so that the Workspace name updates and we get a fresh owner
-      // object so that the formValidation logic keeps working.
-      window.location.reload();
+                <div className="grow font-normal text-element-700">
+                  {isInvitation(elt)
+                    ? elt.inviteEmail
+                    : elt.email || elt.username}
+                </div>
+              </div>
+              <div>
+                {isInvitation(elt) ? (
+                  <Chip size="xs" color="slate">
+                    Invitation {elt.status}
+                  </Chip>
+                ) : (
+                  <Chip
+                    size="xs"
+                    color={COLOR_FOR_ROLE[elt.workspaces[0].role]}
+                    className={
+                      /** Force tailwind to include classes we will need below */
+                      "text-amber-900 text-emerald-900 text-warning-900"
+                    }
+                  >
+                    <span
+                      className={classNames(
+                        "capitalize",
+                        `text-${COLOR_FOR_ROLE[elt.workspaces[0].role]}-900`
+                      )}
+                    >
+                      {elt.workspaces[0].role}
+                    </span>
+                  </Chip>
+                )}
+              </div>
+              <div className="hidden sm:block">
+                <ChevronRightIcon />
+              </div>
+            </div>
+          ))}
+        </div>
+      </>
+    );
+    function isInvitation(
+      arg: MembershipInvitationType | UserType
+    ): arg is MembershipInvitationType {
+      return (arg as MembershipInvitationType).inviteEmail !== undefined;
     }
-  };
+  }
+}
+
+function InviteEmailModal({
+  showModal,
+  onClose,
+  owner,
+}: {
+  showModal: boolean;
+  onClose: () => void;
+  owner: WorkspaceType;
+}) {
+  const [inviteEmail, setInviteEmail] = useState("");
+  const [isSending, setIsSending] = useState(false);
+  const [emailError, setEmailError] = useState("");
+  const [successMessage, setSuccessMessage] = useState("");
+  const { mutate } = useSWRConfig();
+  return (
+    <Modal
+      isOpen={showModal}
+      onClose={onClose}
+      hasChanged={false}
+      title="Invite new users"
+      type="right-side"
+    >
+      <div className="mt-6 flex flex-col gap-6 px-2 text-sm">
+        <Page.P>
+          Send an email to invite a new user to your workspace. They will be
```

Comment:
Wording sounds weird to me but feel free to discard. Proposal to rename from 

> Send an email to invite a new user to your workspace. They will be able to join your workspace by clicking on the link in the email.

to: 

> Invite a new user to your workspace. They will receive an email with a link to  join your workspace.